src/Our.Umbraco.HeadRest/Web/Routing/HeadRestUrlProvider.cs

```diff
@@ -1,39 +1,42 @@
 ﻿using System;
 using System.Collections.Generic;
 using System.Linq;
-using Umbraco.Core.Configuration;
-using Umbraco.Core.Logging;
-using Umbraco.Core;
-using Umbraco.Core.Configuration.UmbracoSettings;
-using Umbraco.Web;
-using Umbraco.Web.Routing;
-using Umbraco.Core.Models.PublishedContent;
+using Umbraco.Cms.Core.Models.PublishedContent;
+using Umbraco.Cms.Core.Routing;
+using Umbraco.Cms.Core.Web;
+using Umbraco.Extensions;
 
 namespace Our.Umbraco.HeadRest.Web.Routing
 {
-    internal class HeadRestUrlProvider : DefaultUrlProvider
+    internal class HeadRestUrlProvider : IUrlProvider
```

Comment:
The `DefaultUrlProvider` has a different constructor in v9 than it does in v10+ which makes it tricy to inherit from without creating different versions of the code.  

Rather than try to have 2 different versions of the code/releases/packages, I've updated the code to implement the `IUrlProvider` instead and injected the `DefaultUrlProvider` so that we can still fall back to use its methods for non-HeadRest URLs.